src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIEmbeddingGenerator.cs

```diff
@@ -160,4 +160,18 @@ private static EmbeddingGeneratorMetadata CreateMetadata(string providerName, st
 
         return openAIOptions;
     }
+
+    private sealed class OpenAIEmbeddingGeneratorMetadata : EmbeddingGeneratorMetadata
+    {
+        private readonly EmbeddingModelMetadata _modelMetadata;
+
+        public OpenAIEmbeddingGeneratorMetadata(string? providerName, Uri? providerUri, string? defaultModelId, int? dimensions)
+            : base(providerName, providerUri, defaultModelId)
+        {
+            _modelMetadata = new() { Dimensions = dimensions };
+        }
+
+        public override Task<EmbeddingModelMetadata> GetModelMetadataAsync(string? modelId = null, CancellationToken cancellationToken = default)
+            => Task.FromResult(_modelMetadata);
```

Comment:
The ctor could also cache the task instead of the metadata.

Reply:
Code was removed.